portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,9 +28,29 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private static final String ENTITY_KIND = "Book";
+  private static final String[] NAME_OF_FIELDS = {"title", "genre", "categories", "author",
+      "language", "description", "infoLink", "pageCount", "publishedDate", "publisher",
+      "maturityRating"};
+  private static final String PAGE_REDIRECT = "/index.html";
+  private static final String TIMESTAMP_PROP = "timeStamp";
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long timeStamp = System.currentTimeMillis();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity bookEntity = new Entity(ENTITY_KIND);
+    for (String field : NAME_OF_FIELDS) {
+      bookEntity.setProperty(field, request.getParameter(field));
+    }
+    bookEntity.setProperty(TIMESTAMP_PROP, timeStamp);
+
+    datastore.put(bookEntity);
+
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.sendRedirect(PAGE_REDIRECT);
   }
```

Comment:
Some of the fields aren't populated - may be the original's lack of content.
![Datastore (without reviews)](https://user-images.githubusercontent.com/51225181/85910346-1d8d3f80-b7d3-11ea-8d96-79f30e791b61.png)

Reply:
that's odd, when I was displaying books I'm fairly certain there were 20. maybe next week we can go through and see if there are any differences between our code to cause a book to get cut off? there might be a for loop or something that accidentally ends at 19 lol.